### ISSUE TITLE
C++ API Build not working

### ISSUE THREAD

**Issue by larry104** [slint-ui/slint#2495]
I follow the exact commands to build sllint but it fails when building the c++ API

`git clone -c core.symlinks=true https://github.com/slint-ui/slint
cargo build
cargo test
mkdir cppbuild && cd cppbuild
cmake -GNinja ..
CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
`

Any advice ?

**Comment by larry104** [comment]
I found the issue. The documation should include the dependency of ninja-build

**Comment by larry104** [comment]
Looks like the documentation 
[https://github.com/slint-ui/slint/blob/master/docs/building.md](url)
does not mention to install ninja-build

**Comment by tronical** [comment]
Thank you for your bug report. You're right, by recommending the use of ninja we should also explain that it needs to be installed.

**Comment by larry104** [comment]
Thanks a lot. Unfortunately, not everyone coming from c++/qt/qmake is familiar with cmake, ninja, rust, cargo etc. so the more complete all dependencies are described the easier it is to "quickly" try out slint and get a "Hello World" example to work and start playing around with it. I should mention that the examples e.g. slint memory game also doesn't compile as described in the tutorial.

**Comment by tronical** [comment]
You’re right. The referencing PR covers the tutorial, too.

**Comment by larry104** [comment]
Another thing worth mentioning in the documentation is that the installation takes almost 30GB. I typically create a Ubuntu 22.04 VM when trying out new things so I have a clean environment. If not sized correctly in the beginning it of cause fails as 30GB's is not really what I would expect for a lightweight tool ;-)

**Comment by ogoffart** [comment]
> installation takes almost 30GB

Yes that's big.  Another reason to recommend the use of the binary package.

(I'm also wondering if there are no way to save lots amount of space by disabling some debug info by default or something)
